WP_User_Query: deprecate dynamic property usage in magic methods

anton-vlasenko opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

<!-- Insert a description of your changes here -->

Deprecates dynamic property usage in the `WP_User_Query` `__get()`, `__set()`, `__isset()`, and `__unset()` methods, using the same strategy done for `WP_List_Table` in changeset https://core.trac.wordpress.org/changeset/56349.

Trac ticket: https://core.trac.wordpress.org/ticket/58897

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

anton-vlasenko committed:
Add doing it wrong as dynamic properties are not supported in PHP 8.2 and above.

hellofromtonya commented:
As we discussed, @anton-vlasenko please update this PR to use the same code strategy done in https://core.trac.wordpress.org/changeset/56349.

anton-vlasenko committed:
Deprecate instead of _doing_it_wrong()

Modifies the changes to match WP_List_Table changes.
See https://core.trac.wordpress.org/changeset/56349.

Changes "defined" to "declared" for better parity to PHP manual.

anton-vlasenko commented:
@hellofromtonya Could you please review this PR? 
I've updated it.
E2E tests are failing, but it seems to be unrelated to the changes in this PR.

hellofromtonya commented on tests/phpunit/tests/user/query.php:
```diff
@@ -13,6 +13,8 @@ class Tests_User_Query extends WP_UnitTestCase {
 
 	protected $user_id;
 
+	private $user_query;
+
```
As most of the tests in this class are setting query variables for their test, I think this isn't needed. 

Instead, let's instantiate in each new test method.

anton-vlasenko replied:
Or maybe other test methods need to be refactored to use the class property instead of local variables.
But I agree that for consistency it's better to use the same approach throughout the test class.

hellofromtonya reviewed (approved):
LGTM 👍 

Failing e2e tests are unrelated to this PR.

Ready for commit.

hellofromtonya commented:
Committed via https://core.trac.wordpress.org/changeset/56353
